internal/core/src/storage/DataCodec.h

```diff
@@ -0,0 +1,105 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <vector>
+#include <memory>
+#include <utility>
+
+#include "storage/Types.h"
+#include "storage/FieldData.h"
+#include "storage/PayloadStream.h"
+
+namespace milvus::storage {
+
+// segment/field meta information corresponding to binlog file data
+struct FieldDataMeta {
+    int64_t collection_id;
+    int64_t partition_id;
+    int64_t segment_id;
+    int64_t field_id;
+};
+
+enum CodecType {
+    InvalidCodecType = 0,
+    InsertDataType = 1,
+    IndexDataType = 2,
+};
+
+class DataCodec {
+ public:
+    explicit DataCodec(std::shared_ptr<FieldData> data, CodecType type) : field_data_(data), codec_type_(type) {
+    }
+
+    virtual ~DataCodec() = default;
+
+    // Serialized data can be written directly to remote or local disk
+    virtual const std::vector<uint8_t>
```

Comment:
should not use "std::vector<uint8_t>", use "std::unique_ptr<uint8_t[]>"

Reply:
We need the size  of the uint8 array